docs/concepts/case-study-snapshot.md

```diff
@@ -0,0 +1,135 @@
+
+## Voting systems are used to calculate the results of a vote
+
+In addition to strategies, each proposal employs one of the following **voting systems**:
+
+- Single choice
+- Approval voting, where a voter can select multiple choices
+- [Quadratic voting](https://en.wikipedia.org/wiki/Quadratic_voting)
+- Weighted voting
+- [and others](https://docs.snapshot.org/proposals/voting-types)
+
+These voting systems are used to calculate the results of a vote based on the voting power. For example, the [following proposal](https://snapshot.org/#/decrypt-media.eth/proposal/QmPDpQSZuokqGpzqtPbmLp1LTqDjbhd6vCeKaSmqEXM9KH)) in the [Decrypt Media](https://decrypt.co/) pace, granted Decrypt DAO members who held Decrypt NFTs the right to vote on what content they will write next.
+
+![Decrypt content proposal](./images/case-studies/img-snapshot-decrypt-proposal.png)
+
+## How Snapshot uses IPFS
+
+Snapshot uses IPFS to make the whole voting process fully transparent and auditable. In practice, every space, proposal, vote, and user action is added to IPFS and has [CID](/concepts/content-addressing/).
+
+Additionally, the Snapshot UI is also [available on IPFS](https://bafybeihzjoqahhgrhnsksyfubnlmjvkt66aliodeicywwtofodeuo2icde.ipfs.dweb.link/).
+
+To understand how Snapshot uses IPFS, it's useful to understand how the whole architecture was designed. Snapshot is a hybrid app combining design patterns common to Web2 and Web3 apps, and built based on the three-tier architecture:
+
+- **Presentation tier:** [The Snapshot UI](https://github.com/snapshot-labs/snapshot). The UI is also available over IPFS (see the `x-ipfs-path` header when making an HTTP request to `https://snapshot.org`.)
+- **Logic tier:** The [snapshot-hub](https://github.com/snapshot-labs/snapshot-hub) node.js server that exposes a GraphQL API.
+- **Data tier:** A combination of a MySQL database for indexing and querying and IPFS as its storage layer for space configuration, proposals, user actions, and votes.
+
+> **Note:** Even though MySQL is used for querying, all data stored on MySQL is available on IPFS. [Tables in the database schema](https://github.com/snapshot-labs/snapshot-hub/blob/master/src/helpers/schema.sql) have an `ipfs` field holding the IPFS CID.
+
+![Snapshot architecture](./images/case-studies/img-snapshot-architecture.png)
+
+### Uploading to IPFS pinning services with Pineapple
+
+To make it easier to persist data to IPFS, Snapshot built a microservice called [Pineapple](https://github.com/snapshot-labs/pineapple) and a client library called [pineapple.js](https://github.com/snapshot-labs/pineapple.js). Together they abstract the details of uploading content to multiple IPFS pinning services.
```

Comment:
IIRC, it was because of variable latency uploading and to ensure better availability of the CIDs on the IPFS network.

This is why they implemented an async race uploading to the 5 services so that they can respond from the microservice as soon as the first successful upload completes. 

The complexity is pretty minimal:
- Provider config https://github.com/snapshot-labs/pineapple/tree/master/src/providers
- Race https://github.com/snapshot-labs/pineapple/blob/f1fca1a34adefab1c96cdd426d893ee88322b170/src/rpc.ts#L19-L25